### PR TITLE
{geo}[SYSTEM] SNAP-ESA v8.0 and python interface with Python-2.7.18/GCCcore-10.2.0

### PR DESCRIPTION
**Pull request by lexming**
Depends on ~~https://github.com/easybuilders/easybuild-easyconfigs/pull/13141~~

This PR has the particularity that the Python module for SNAP-ESA `snappy` is installed using a tool from `SNAP-ESA` instead of using some sources. Therefore, I made a creative use of `Bundle` for `SNAP-ESA-python`.

On top of that, several commands of SNAP fail to properly terminate themselves after the execution is complete. They just hang forever. This issue is known by the devs but there is no fix (probably low priority because `ctrl-C` is easy enough :laughing: ). So I resorted to parse the output of those tools and  _manually_ kill them once they are done.

The update of SNAP's modules is disabled by default because there is no way to control which versions get installed with it. It always downloads the latests versions from their repos. Use at your own risk.

### CODE REVIEW

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node381.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/5b0ace71e6d5c75bb4496b2608e99de5 for a full test report.

**Review by jfgrimm — COMMENTED** [review]
As decided in issue #16330, we have deprecated the use of `True` to signify a system-toolchain dependency (#16384), in favour of the more intuitive `SYSTEM` template constant. Due to the change in the test suite, please run `eb --sync-pr-with-develop 13198` and update the PR to use `SYSTEM` instead.

**Review by boegel — CHANGES_REQUESTED** [review]
see https://github.com/lexming/easybuild-easyconfigs/pull/2

File: easybuild/easyconfigs/s/SNAP-ESA-python/SNAP-ESA-python-8.0-GCCcore-10.2.0-Java-1.8.0_292-OpenJDK-Python-2.7.18.eb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+local_javasuffix = '-Java-%(javaver)s-OpenJDK'
+versionsuffix = local_javasuffix + '-Python-%(pyver)s'
+
+homepage = 'https://step.esa.int/main/toolboxes/snap/'
+description = "Python interface to the Sentinel Application Platform (SNAP) API"
+
+toolchain = {'name': 'GCCcore', 'version': '10.2.0'}
+
+dependencies = [
+    ('Java', '1.8.0_292', '-OpenJDK', SYSTEM),
```
[the remark this answers is not in the document]
Comment (lexming) [reply easybuild/easyconfigs/s/SNAP-ESA-python/SNAP-ESA-python-8.0-GCCcore-10.2.0-Java-1.8.0_292-OpenJDK-Python-2.7.18.eb]: SNAP developers recommend OpenJDK, but it is not a hard requirement so we can switch to the wrapper and let the installer choose. I updated the comment to clarify that an OpenJDK is recommended.

File: easybuild/easyconfigs/s/SNAP-ESA-python/SNAP-ESA-python-8.0-GCCcore-10.2.0-Java-1.8.0_292-OpenJDK-Python-2.7.18.eb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+sanity_check_commands = ["python -c 'import snappy'"]
+
+modextrapaths = {'PYTHONPATH': [local_pysite]}
+
+local_javaopts = " -Dsnap.pythonExecutable=python"
+local_javaopts += " -Dsnap.pythonModuleDir=%s" % local_pyinstalldir
+
+modluafooter = 'setenv("JAVA_TOOL_OPTIONS", os.getenv("JAVA_TOOL_OPTIONS").."%s")' % local_javaopts
```
[the remark this answers is not in the document]
Comment (lexming) [reply easybuild/easyconfigs/s/SNAP-ESA-python/SNAP-ESA-python-8.0-GCCcore-10.2.0-Java-1.8.0_292-OpenJDK-Python-2.7.18.eb]: good catch, thanks for the fix :+1:

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3130.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/6e3124a6fed0d259a5c3bf1a7ce2d41f for a full test report.

**Comment by boegel** [comment]
Going in, thanks @lexming!

**Comment by boegel** [comment]
Going in, thanks @lexming!

**Pull request merged**